doc/user-guide/push-notifications/Push-notifications.md

```diff
@@ -0,0 +1,155 @@
+node (the most common setup!). Or, for the best performance, you might just skip that PubSub node
+altogether. While the whole setup can be incredibly extensible, we see the following straightforward
+uses of it.
+
+### XEP-0357 compliant with local PubSub
+
+ ![](PushNotifications-MIM_with_local_pubsub.png)
+
+This is, historically, the most common setup.
+It allows your clients to enable push notifications via a local PubSub,
+and the IQ stanza is routed internally.
+
+A direct connection to a push service (e.g. MongoosePush) must be configured on the same MongooseIM
+node, check out [this][WithMongoosePush] tutorial on how to setup [MongoosePush][].
+
+```erlang
+{mod_pubsub, [{plugins, [<<"push">>]}]}, % mandatory minimal config
+{mod_event_pusher, [
+    {backends, [
+        {push, [
+            {backend, mnesia}, % optional
+            {wpool, [{workers, 200}]}, % optional
+            {plugin_module, mod_event_pusher_push_plugin_defaults} % optional
+        ]}
+    ]}
+]}
+```
+
+#### Advantages
+* Completely XEP-0357 compliant, and therefore compatible with any compliant 3rd party client library
+* No need to have two different servers
+
+#### Drawbacks
+* Less efficient (PubSub has a considerable impact on heavily loaded systems)
+* More load within a single node
+* Harder to customise
+
+### MongooseIM as a PubSub-less XMPP server
+
+ ![](PushNotifications-no_pubsub.png)
+
+PubSub is completely bypassed and clients don't need to create a push node — if they attempt to
+do so, and PubSub is not configured, the server would respond with an error stanza.
+They only have to provide the virtual PubSub address in the enable stanza, and node name can be
+anything unique. In order to ensure uniqueness the APNS/FCM token can be used. Note that the token
+must be provided [as a publish option](https://xmpp.org/extensions/xep-0357.html#example-9) anyway.
+
+A direct connection to a push service (e.g. MongoosePush) must be configured on the same MongooseIM
+node, check out [this][WithMongoosePush] tutorial on how to setup [MongoosePush][].
+
+```Erlang
+{mod_event_pusher, [
+    {backends, [
+        {push, [
+            {backend, mnesia}, % optional
+            {wpool, [{workers, 200}]}, % optional
+            {plugin_module, mod_event_pusher_push_plugin_defaults}, % optional
+            {virtual_pubsub_hosts, ["virtual.@HOST@"]}
+        ]}
+    ]}
+]}
+```
+
+#### Advantages
+* No need to use PubSub at all
+* More efficient (PubSub has a considerable impact on heavily loaded systems)
+* Simpler client-side usage — Read about the [client side configuration here][client-side].
+
+#### Drawbacks
+* If the client application is build to create the push PubSub node, this might require a migration
+  for such client — as he attempts to create the node, the server will answer with an IQ error
+  stanza if the domain is not shadowed (using `virtual_pubsub_hosts`). If migrating the client side
```

Comment:
this is completely incorrect and must be changed. is pubsub is not configured server responds on node creation request with IQ error even if domain is shadowed! shadowing is not solving the this problem hence the need for migration tricks!